test/arbitrage.test.js

```diff
@@ -182,32 +266,150 @@ test('all no true', t => {
+  var act = r.setdiff(a,b)
+  var exp = [3]
+  t.true(r.all(r.is_equal(act, exp)))
+})
 
-test.todo('is_equal_cols 1')
-test.todo('is_equal_cols throws error with mismatched lengths')
+test('setdiff a subset of b', t => {
+  var a = [ 1,2 ]
+  var b = [ 1,2,3 ]
+  var act = r.setdiff(a,b)
+  var exp = []
+  t.true(r.all(r.is_equal(exp, act)))
+})
 
+test('setdiff a and b disjoint', t => {
+  var a = [ 1,2 ]
+  var b = [ 3,4 ]
+  var act = r.setdiff(a,b)
+  var exp = [1,2]
+  t.true(r.all(r.is_equal(exp, act)))
+})
+
+test('intersection b subset of a', t => {
+  var a = [ 1,2,3 ]
+  var b = [ 1,2 ]
+  var act = r.intersection(a,b)
+  var exp = [1,2]
+  t.true(r.all(r.is_equal(exp, act)))
+})
 
-test.todo('setdiff 1')
-test.todo('setdiff b subset of a')
-test.todo('setdiff a subset of b')
-test.todo('setdiff a and b disjoint')
+test('intersection a subset of b', t => {
+  var a = [ 1,2 ]
+  var b = [ 1,2,3 ]
+  var act = r.intersection(a,b)
+  var exp = [1,2]
+  t.true(r.all(r.is_equal(act, exp)))
+})
+
+test('intersection a and b disjoint', t => {
+  var a = [ 1,2 ]
+  var b = [ 3,4 ]
+  var act = r.intersection(a,b)
+  var exp = []
+  t.true(r.all(r.is_equal(act, exp)))
+})
+
+test('intersection a = b', t => {
+  var a = [ 1,2 ]
+  var b = [ 1,2 ]
+  var act = r.intersection(a,b)
+  var exp = [1,2]
+  t.true(r.all(r.is_equal(act, exp)))
+})
+
+test('union b subset of a', t => {
+  var a = [ 1,2,3 ]
+  var b = [ 1,2 ]
+  var act = r.union(a,b)
+  var exp = new Set([1,2,3])
+  t.true(r.all(r.is_equal(act, exp)))
+})
+
+test('union a subset of b', t => {
+  var a = [ 1,2 ]
+  var b = [ 1,2,3 ]
+  var act = r.union(a,b)
+  var exp = new Set([1,2,3])
+  t.true(r.all(r.is_equal(act, exp)))
+})
+
+test('union a and b disjoint', t => {
+  var a = [ 1,2 ]
+  var b = [ 3,4 ]
+  var act = r.union(a,b)
+  var exp = new Set([1,2,3,4])
+  t.true(r.all(r.is_equal(act, exp)))
+})
+
+test('union a = b', t => {
+  var a = [ 1,2 ]
+  var b = [ 1,2 ]
+  var act = r.union(a,b)
+  var exp = new Set([1,2])
+  t.true(r.all(r.is_equal(act, exp)))
+})
 
-test.todo('intersection b subset of a')
-test.todo('intersection a subset of b')
-test.todo('intersection a and b disjoint')
-test.todo('intersection a = b')
+test('within some x in xs', t => {
+  var xs = [ 1,2,3 ]
+  var x = [ 1,2 ]
+  var act = r.within(x,xs)
+  var exp = [true,true]
+  t.true(r.all(r.is_equal(act, exp)))
+})
 
-test.todo('union b subset of a')
-test.todo('union a subset of b')
-test.todo('union a and b disjoint')
-test.todo('union a = b')
+test('within all x in xs', t => {
+  var xs = [ 1,2,3 ]
+  var x = [ 1,2,3 ]
+  var act = r.within(x,xs)
+  var exp = [true,true,true]
+  t.true(r.all(r.is_equal(act, exp)))
+})
 
-test.todo('within some x in xs')
-test.todo('within all x in xs')
-test.todo('within no x not in xs')
+test('within no x not in xs', t => {
+  var xs = [ 1,2,3 ]
+  var x  = [ 4,5 ]
+  var act = r.within(x,xs)
+  var exp = [false,false]
+  t.true(r.all(r.is_equal(act, exp)))
+})
```

Comment:
Are these the only cases? It seems to me that you're missing at least one.

Reply:
I've added more `todo` tests